### PR TITLE
Feat/auth page

### PR DESCRIPTION
**Pull request by biskuvit**
Added auth page for common authentication forms.

Please check all items below before review.

-   [x] Corresponding issues are created/updated or not needed
-   [x] Docs are updated/provided or not needed
-   [x] Examples are updated/provided or not needed
-   [x] TypeScript definitions are updated/provided or not needed
-   [x] Tests are updated/provided or not needed
-   [x] Changesets are provided or not needed

### CODE REVIEW

File: packages/react-router-v6/src/routeProvider.tsx [hunk 1]
```diff
@@ -267,11 +268,27 @@ export const RouteProvider = () => {
             ))}
             <Route
                 path="/"
-                element={LoginPage ? <LoginPage /> : <DefaultLoginPage />}
+                element={
+                    AuthPage ? (
```
Comment (omeraplak) [inline packages/react-router-v6/src/routeProvider.tsx:272]: Can we convert this logic to function? Looks like we'll have to do the same on all routers

File: packages/core/src/components/pages/auth/components/login/index.tsx [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+
+    const translate = useTranslate();
+
+    const { mutate: login } = useLogin<ILoginForm>();
+
+    const renderLink = (link: React.ReactNode, text?: string) => {
+        if (link) {
+            if (typeof link === "string") {
+                return <Link to={link}>{text}</Link>;
+            } else return link;
```
Comment (yildirayunlu) [inline packages/core/src/components/pages/auth/components/login/index.tsx:24]:
```suggestion
            if (typeof link === "string") {
                return <Link to={link}>{text}</Link>;
            }
            return link;
```

File: packages/core/src/components/pages/auth/components/register/index.tsx [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+                        required
+                        size={20}
+                        value={password}
+                        onChange={(e) => setPassword(e.target.value)}
+                    />
+                    {translate(
+                        "pages.register.confirmPassword",
+                        undefined,
+                        "Confirm Password",
+                    )}
```
Comment (yildirayunlu) [inline packages/core/src/components/pages/auth/components/register/index.tsx:82]: should be in the `<label>` tag.

File: packages/core/src/components/pages/auth/components/resetPassword/index.tsx [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+    const [email, setEmail] = useState("");
+
+    const { mutate: resetPassword } = useResetPassword<IResetPasswordForm>();
+
+    const renderLink = (link: React.ReactNode, text?: string) => {
+        if (link) {
+            if (typeof link === "string") {
+                return <Link to={link}>{text}</Link>;
+            } else return link;
```
Comment (yildirayunlu) [inline packages/core/src/components/pages/auth/components/resetPassword/index.tsx:20]:
```suggestion
            if (typeof link === "string") {
                return <Link to={link}>{text}</Link>;
            } 
            return link;
```

File: packages/core/src/components/pages/auth/components/updatePassword/index.tsx [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+                        type="password"
+                        required
+                        size={20}
+                        value={password}
+                        onChange={(e) => setPassword(e.target.value)}
+                    />
+                    {translate(
+                        "pages.updatePassword.newPassword",
+                        "New Password",
+                    )}
```
Comment (yildirayunlu) [inline packages/core/src/components/pages/auth/components/updatePassword/index.tsx:65]: should be in the `<label>` tag.

File: packages/core/src/components/pages/auth/components/updatePassword/index.tsx [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+                    <input
+                        type="password"
+                        required
+                        size={20}
+                        value={newpPassword}
+                        onChange={(e) => setNewPassword(e.target.value)}
+                    />
+                    {translate(
```
Comment (yildirayunlu) [inline packages/core/src/components/pages/auth/components/updatePassword/index.tsx:73]: should be in the `<label>` tag.

**Review by omeraplak — APPROVED** [review]

**Review by omeraplak — APPROVED** [review]

**Review by yildirayunlu — APPROVED** [review]

File: packages/core/src/hooks/auth/useLogin/index.ts [hunk 7]
```diff
@@ -30,7 +30,9 @@ export const useLogin = <TVariables = {}>(): UseMutationResult<
     const { search } = useLocation();
     const { close, open } = useNotification();
 
-    const { to } = qs.parse(search?.substring(1));
+    const { to } = qs.parse(search, {
```
Comment (omeraplak) [inline packages/core/src/hooks/auth/useLogin/index.ts:33]: 🚀

**Review by omeraplak — APPROVED** [review]

File: .changeset/real-planes-float.md [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+
+🎉 Added `AuthPage` to the `refine` app. This page is used to login, register, reset password and update password. Login page is default page and old `LoginPage` prop is deprecated.
+
+# New Auth Hooks
+
+📌 Added `useRegister` hook. This hook is used to register new user. `userRegister` falls into register function of [`AuthProvider`](https://refine.dev/docs/core/providers/auth-provider/).
+
+📌 Added `useResetPassword` hook. This hook is used to reset password. `userResetPassword` falls into `resetPassword` function of [`AuthProvider`](https://refine.dev/docs/core/providers/auth-provider/).
+
+📌 Added `useUpdatePassword` hook. This hook is used to update password. `userUpdatePassword` falls into `updatePassword` function of [`AuthProvider`](https://refine.dev/docs/core/providers/auth-provider/).
```
Comment (omeraplak) [inline .changeset/real-planes-float.md:17]:
```suggestion
📌 Added `useUpdatePassword` hook. This hook is used to update password. `useUpdatePassword` falls into `updatePassword` function of [`AuthProvider`](https://refine.dev/docs/core/providers/auth-provider/).
```

File: examples/fineFoods/admin/antd/src/pages/dashboard/index.tsx [hunk 9]
```diff
@@ -83,7 +83,7 @@ export const DashboardPage: React.FC = () => {
                         </Text>
                     }
                 >
-                    <DeliveryMap />
+                    {/* <DeliveryMap /> */}
```
Comment (omeraplak) [inline examples/fineFoods/admin/antd/src/pages/dashboard/index.tsx:86]:
```suggestion
                    <DeliveryMap />
```

File: packages/core/src/components/pages/auth/components/login/index.tsx [hunk 10]
```diff
@@ -0,0 +1,139 @@
...
+                        alignItems: "center",
+                        justifyContent: "center",
+                        marginBottom: "1rem",
+                    }}
+                >
+                    <button
+                        onClick={() =>
+                            login({
+                                email,
+                                password,
```
[the remark this answers is not in the document]
Comment (omeraplak) [reply packages/core/src/components/pages/auth/components/login/index.tsx:46]: Do we need email and password in the provider login?

File: packages/core/src/components/pages/auth/components/register/index.tsx [hunk 11]
```diff
@@ -0,0 +1,123 @@
...
+                                "pages.login.resetPassword",
+                                "Forgot your password?",
+                            ),
+                        )}
+                    {registerLink ? (
+                        renderLink(
+                            registerLink,
+                            translate("pages.register.button", "Register"),
+                        )
+                    ) : (
```
Comment (omeraplak) [inline packages/core/src/components/pages/auth/components/register/index.tsx:107]: How exactly is this used?

File: packages/core/src/components/pages/auth/components/login/index.tsx [hunk 12]
```diff
@@ -0,0 +1,139 @@
...
+                        value={password}
+                        onChange={(e) => setPassword(e.target.value)}
+                    />
+                    <br />
+                    {loginLink ? (
+                        renderLink(
+                            loginLink,
+                            translate("pages.login.signup", "Login"),
+                        )
+                    ) : (
```
Comment (omeraplak) [inline packages/core/src/components/pages/auth/components/login/index.tsx:119]: How exactly is this used?

**Review by omeraplak — CHANGES_REQUESTED** [review]

File: examples/authentication/headless/index.html [hunk 13]
```diff
@@ -0,0 +1,13 @@
+<!DOCTYPE html>
+<html lang="en">
+    <head>
+        <meta charset="UTF-8" />
+        <link rel="icon" href="/src/favicon.ico" />
+        <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+        <title>refine headless authenticationexample</title>
```
Comment (salihozdemir) [inline examples/authentication/headless/index.html:7]:
```suggestion
        <title>refine headless authentication example</title>
```

File: examples/authentication/headless/src/pages/posts/create.tsx [hunk 14]
```diff
@@ -0,0 +1,100 @@
+import React, { useState } from "react";
+import { IResourceComponentsProps } from "@pankod/refine-core";
+
+import { Create, Form, Input, Select } from "@pankod/refine-antd";
+
+import { useForm, useSelect } from "@pankod/refine-antd";
```
Comment (ozkalai) [inline examples/authentication/headless/src/pages/posts/create.tsx:6]: imports can be merge

File: packages/core/src/components/containers/refine/index.tsx [hunk 15]
```diff
@@ -58,7 +58,9 @@ export interface RefineProps {
     resources?: ResourceProps[];
     i18nProvider?: I18nProvider;
     catchAll?: React.ReactNode;
+    /** @deprecated Login page to AuthPage */
```
Comment (salihozdemir) [inline packages/core/src/components/containers/refine/index.tsx:61]:
```suggestion
    /** @deprecated LoginPage deprecated. Use AuthPage instead. */
```

File: packages/core/src/components/pages/auth/components/resetPassword/index.tsx [hunk 16]
```diff
@@ -0,0 +1,78 @@
+import { useTranslate, useRouterContext, useResetPassword } from "@hooks";
```
Comment (ozkalai) [inline packages/core/src/components/pages/auth/components/resetPassword/index.tsx:1]: import order can be arrange

File: examples/authentication/headless/src/pages/posts/create.tsx (hunk 14 above)
Comment (ozkalai) [reply examples/authentication/headless/src/pages/posts/create.tsx:6]: antd elements should removed from headless examples

File: examples/authentication/headless/src/pages/posts/edit.tsx [hunk 17]
```diff
@@ -0,0 +1,102 @@
+import React, { useState } from "react";
+import { IResourceComponentsProps } from "@pankod/refine-core";
+
+import { Edit, Form, Input, Select } from "@pankod/refine-antd";
+
+import { useForm, useSelect } from "@pankod/refine-antd";
```
Comment (ozkalai) [inline examples/authentication/headless/src/pages/posts/edit.tsx:6]: antd elements should remove from headless examples

File: examples/authentication/headless/src/pages/posts/list.tsx [hunk 18]
```diff
@@ -0,0 +1,124 @@
+import {
+    IResourceComponentsProps,
+    useMany,
+    getDefaultFilter,
+} from "@pankod/refine-core";
+
+import {
```
Comment (ozkalai) [inline examples/authentication/headless/src/pages/posts/list.tsx:7]: antd elements should remove from headless example

File: examples/authentication/headless/src/pages/posts/show.tsx [hunk 19]
```diff
@@ -0,0 +1,40 @@
+import { useShow, IResourceComponentsProps, useOne } from "@pankod/refine-core";
+
+import { Show, Typography, MarkdownField } from "@pankod/refine-antd";
```
Comment (ozkalai) [inline examples/authentication/headless/src/pages/posts/show.tsx:3]: antd elements should remove from headless example

**Review by salihozdemir — APPROVED** [review]

File: packages/core/src/components/pages/auth/components/updatePassword/index.tsx [hunk 20]
```diff
@@ -0,0 +1,115 @@
...
+    backLink,
+    updatePasswordLink,
+}) => {
+    const translate = useTranslate();
+    const { Link } = useRouterContext();
+
+    const { mutate: updatePassword } = useUpdatePassword<IUpdatePasswordForm>();
+
+    const [password, setPassword] = useState("");
+    const [newpPassword, setNewPassword] = useState("");
```
Comment (ozkalai) [inline packages/core/src/components/pages/auth/components/updatePassword/index.tsx:16]:
```suggestion
    const [newPassword, setNewPassword] = useState("");
```
I am not sure the name of state but it has typo

File: packages/core/src/components/pages/auth/index.tsx [hunk 21]
```diff
@@ -0,0 +1,108 @@
+import React from "react";
+
+import { Login, Register, ResetPassword, UpdatePassword } from "./components";
+
+export interface ILoginForm {
```
Comment (ozkalai) [inline packages/core/src/components/pages/auth/index.tsx:5]: Should login form entries be optional or mandatory, especially passwords?

**Review by omeraplak — APPROVED** [review]

**Pull request merged**